core/packet.c

```diff
@@ -192,93 +182,102 @@ void lwm2m_handle_packet(lwm2m_context_t * contextP,
+                    }
 
-                    coap_set_header_block2(response, 0, new_offset!=-1, REST_MAX_CHUNK_SIZE);
-                    coap_set_payload(response, response->payload, MIN(response->payload_len, REST_MAX_CHUNK_SIZE));
-                } /* if (blockwise request) */
+                    if ( IS_OPTION(message,
+                            COAP_OPTION_BLOCK1) && response->code < BAD_REQUEST_4_00 && !IS_OPTION(response, COAP_OPTION_BLOCK1))
+                    {
+                        LOG("Block1 NOT IMPLEMENTED\n");
 
-                coap_error_code = message_send(contextP, response, fromSessionH);
+                        coap_error_code = NOT_IMPLEMENTED_5_01;
+                        coap_error_message = "NoBlock1Support";
+                    }
 
-                lwm2m_free(response->payload);
-                response->payload = NULL;
-                response->payload_len = 0;
-            }
-            else
-            {
-            	coap_error_code = message_send(contextP, response, fromSessionH);
+                    coap_error_code = message_send(contextP, response, fromSessionH);
+#ifdef LWM2M_CLIENT_MODE
+                    if (notify_change) {
+                        lwm2m_resource_value_changed(contextP, uriP);
+                    }
+#endif
+                }
+                lwm2m_free(uriP);
+                lwm2m_free(payload);
             }
         }
         else
         {
-            /* Responses */
-            lwm2m_transaction_t * transaction;
-
-            if (message->type==COAP_TYPE_ACK)
-            {
-                LOG("Received ACK\n");
-            }
-            else if (message->type==COAP_TYPE_RST)
-            {
-                LOG("Received RST\n");
-                /* Cancel possible subscriptions. */
-                handle_reset(contextP, fromSessionH, message);
-            }
+            if (message->type == COAP_TYPE_NON || message->type == COAP_TYPE_CON ) {
 
 #ifdef LWM2M_SERVER_MODE
-            if ( (message->code == COAP_204_CHANGED || message->code == COAP_205_CONTENT)
-             && IS_OPTION(message, COAP_OPTION_OBSERVE))
-            {
-                handle_observe_notify(contextP, fromSessionH, message);
-            }
-            else
+                if ( (message->code == COAP_204_CHANGED || message->code == COAP_205_CONTENT)
+                        && IS_OPTION(message, COAP_OPTION_OBSERVE))
+                {
+                    handle_observe_notify(contextP, fromSessionH, message);
+                }
+                else
 #endif
-            {
+#ifdef LWM2M_CLIENT_MODE
+                if (IS_OPTION(message, COAP_OPTION_LOCATION_PATH)) {
+                    multi_option_t *location_path = message->location_path;
+                    if ((location_path->len == 2) && (memcmp(location_path->data, "rd", 2) == 0)) {
+                        handle_registration_response(contextP, fromSessionH, message);
```

Comment:
I'm not sure to understand this : are you expecting a registration response in a CON or NON message ?